### ISSUE TITLE
An `unless conflict else` inside a trigger fails with "object type or alias does not exist" error

### ISSUE THREAD

**Issue by TheHyphen** [edgedb/edgedb#5798]
- EdgeDB Version: 3.1+a33a828
- EdgeDB CLI Version: 3.4.0+4d95a2c
- OS Version: MacOS 12.3

Steps to Reproduce:

An `on conflict` with an `else` clause in a trigger makes the `edgedb migrate --dev-mode` fail with the error
> object type or alias 'Stock' does not exist

Schema:
```
module default {
    type Product {
        required name: str;
    }

    type Stock {
        required product: Product {
            constraint exclusive;
        };
        required quantity: int32;
    }

    type IncomingStock {
        required product: Product;
        required quantity: int32;

        trigger StockCreate after insert for each do (
            (
                insert Stock {
                    product := __new__.product,
                    quantity := __new__.quantity
                }
                unless conflict on .product
                else (
                    update Stock set { quantity := .quantity + __new__.quantity }
                )
            )
        );
    }
}
```

Full error message:
```
❯ edgedb migrate --dev-mode
error: object type or alias 'Stock' does not exist
   ┌─ /Users/personal/projects/edgedb/dbschema/default.esdl:17:54
   │
17 │           trigger StockCreate after insert for each do (
   │ ╭──────────────────────────────────────────────────────^
18 │ │             insert Stock {
19 │ │                 product := __new__.product,
20 │ │                 quantity := __new__.quantity
   · │
25 │ │             )
26 │ │         );
   │ ╰─────────^ error

edgedb error: cannot proceed until .esdl files are fixed
```

This error does not occur without the `else` clause in the schema.

**Comment by mkhlsvn** [comment]
This annoying bug still actual not only for 3.4+301ba34 but also for 5.0-dev.7931+b66b7b5

```
error: object type or alias 'DayClosureEvent' does not exist
     ┌─ C:\Users\...\dbschema\default-v1.33.1.esdl:1621:13
     │
1621 │ ╭             FOR d IN range_unpack(range(func::getLocalDate(new.date), func::getCurrentDate())) UNION (
1622 │ │                 INSERT LoanAccruals {date:= d, loan:= <Loan>__new__.id,
1623 │ │                     # error: object type or alias 'DayClosureEvent' does not exist
1624 │ │                     # https://github.com/edgedb/edgedb/issues/5798
1625 │ │                     dayClosure:= (INSERT DayClosureEvent {date:= d, closureType:= EventType.AUTOMATIC} UNLESS CONFLICT ON (.tenant, .date) ELSE (SELECT DayClosureEvent))})
     │ ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────^ error
```

**Comment by msullivan** [comment]
This will be fixed in https://github.com/edgedb/edgedb/pull/6297.

As a workaround until then, try fully qualifying the name in the else branch:
```
update default::Stock set ...
```

**Comment by msullivan** [comment]
Fixed by  #6297
